src/containers/forms/PlanForm/helpers.ts

```diff
@@ -150,13 +150,16 @@ export function extractActivityForForm(activityObj: PlanActivity): PlanActivityF
     actionTitle: activityObj.action.title || '',
     goalDescription: activityObj.goal.description || '',
     goalDue:
-      activityObj.goal.target[0].due && activityObj.goal.target[0].due !== ''
+      activityObj.goal.target &&
+      activityObj.goal.target[0].due &&
+      activityObj.goal.target[0].due !== ''
         ? parseISO(`${activityObj.goal.target[0].due}${DEFAULT_TIME}`)
         : moment()
             .add(DEFAULT_ACTIVITY_DURATION_DAYS, 'days')
             .toDate(),
     goalPriority: activityObj.goal.priority || goalPriorities[1],
-    goalValue: activityObj.goal.target[0].detail.detailQuantity.value || 0,
+    goalValue:
+      (activityObj.goal.target && activityObj.goal.target[0].detail.detailQuantity.value) || 0,
```

Comment:
Could we use the default value of IRS plans from settings, instead of 0?

Reply:
@moshthepitt not sure what you mean by `the default value of IRS plans from settings`, do you mean the variable `ZERO`?

Reply:
I mean this: https://github.com/onaio/reveal-frontend/blob/master/src/configs/settings.ts#L280  - 0 is actually invalid for that field :(

Reply:
oooh i see, sorry i thought you meant the `[0]` not the `|| 0`, I will replace the later with the default value. :)